### PR TITLE
kill the watch script if the .packages file changes

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/build/issues/777

### CODE REVIEW

File: build_runner/lib/src/generate/watch_impl.dart [hunk 1]
```diff
@@ -230,6 +232,12 @@ class WatchImpl implements BuildState {
   /// Checks if we should skip a watch event for this [change].
   bool _shouldProcess(AssetChange change) {
     assert(_assetGraph != null);
+    if (_isRootPackagesFile(change.id)) {
+      _fatalBuildCompleter.complete();
```
Comment (natebosch) [inline build_runner/lib/src/generate/watch_impl.dart]: Is there a way we can structure this without adding side effects to this method?

Comment (jakemac53) [reply build_runner/lib/src/generate/watch_impl.dart]: I am looking into it, but without setting up an additional watcher it is not clear what the best way would be.

Comment (jakemac53) [reply build_runner/lib/src/generate/watch_impl.dart]: I moved it up earlier so at least it isn't so hidden inside of `_shouldProcess`, lmk what you think.

Comment (natebosch) [reply build_runner/lib/src/generate/watch_impl.dart]: I like the new location inside the `asyncMap` call.

**Pull request merged**